docs/ccv1_image_security_design.md

```diff
@@ -0,0 +1,403 @@
+
+## Deployment of protected container image
+
+In the confidential container V1 design, security functions will be performed when pulling the container image, including checking the registry allowlist, verifying the container image signature, and decrypting the container image layer. The high-level process is as follows:
+
+1. Obtain the policy set matching the image from the policy file. If it is rejected, the pull action will be terminated directly; If it is unconditional acceptance, skip the second and third steps; If it is a signature verification requirement, proceed to the following steps
+
+2. (if needed in policy) Download the signature claim file of the container image according to the configuration of the sigstore file.
+
+3. (if needed in policy) Verify the signature of the container image according to the configuration of the policy.
+
+4. Pull the container image layer by layer. If an encrypted container image layer is encrypted, decrypt it.
+
+After the image is pulled, the subsequent deployment process is consistent with that of the normal container image.
+
+### policy.json
+
+policy.json is a necessary configuration file for deploying images in the confidential container V1 design. Its location is specified in the /etc/containers/ directory of boot image rootfs. Its main functions are as follows:
+
+- Declare a allowlist of container images with "transport - scope" granularity
+
+- Specify the public key which should to be used when verifying the image signature
+
+In the current design of ccv1, the following two types of transport are supported:
+
+- "docker": used to match the image in the container registry that implements the docker registry HTTP API V2 protocol.
+
+- "dir": used to match images located in the local file system.
+
+For the "docker" transport type, the scope can be a docker reference named in a fully expanded form, and tag or digest can be additionally specified for exact matching, for example: `docker.io/library/busybox:latest`. Generic matching is also supported:
+
+- repository namespace: for example, `docker.io/openshift`, which matches the images as `openshift/hello-openshift` and `openshift/origin`.
+
+- repository host: for example, `docker.io`, matches all images from the docker.io repository.
+
+- subdomain name matching: such as `*.temporary-project.example.com`. Note: wildcards can only appear in the top sub domain name. Wildcards like `example.*.com` are not allowed to be specified.
+
+For the "dir" transport type, the scope format is the directory path. The path must be absolute and contain no symbolic links. The default scope is "".
+
+The policy file must be generated and maintained by the user. Additional security means need to be used to ensure that the policy file will not be modified by malicious entities during storage and transmission. If users have high requirements for security level, they even need to use additional means to ensure the confidentiality of policy files.
```

Comment:
Makes me wonder if we should include a digest of the policy.json file as part of what we send during attestation.

Reply:
Strike the above, we do when we measure the rootfs anyway.

Reply:
Yes, policy.json, as the content of boot image, will indeed be measured during the attestation and sent as evidence for attestation.